### PR TITLE
This vendor is suspicious!

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
- emagging a vendor now makes it hostile like in the rampant brand intelligence event
it doesn't spread to other vendors though.
## Why It's Good For The Game
I think it's really funny and fits the emag action, also a great method to make death traps.
## Changelog
:cl: grungussuss
add: vendors can now be emaged to make them hostile
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I'm a little wary of this as an emag function given how you'd be able to pretty rapidly do it to every vending machine you walk past, but on the other hand while they're very harmful they're also reasonably easy to stop if anyone has tools

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -1189,9 +1189,13 @@ GLOBAL_LIST_EMPTY(vending_machines_to_restock)
 	if(obj_flags & EMAGGED)
 		return FALSE
 	obj_flags |= EMAGGED
-	balloon_alert(user, "product lock disabled")
+	addtimer(CALLBACK(src, PROC_REF(become_hostile)), rand(10, 30))
```
Comment (Ghommie) [inline code/modules/vending/_vending.dm:1192]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(become_hostile)), rand(1 SECONDS, 3 SECONDS))
```

**Comment by Loafin34** [comment]
pssh emags are barely seen anymore and this is way better than just "contraband". you can get contraband easy, and the emag's supposed to wreck havoc!

**Comment by SomeoneYouProbablyKnow** [comment]
The big danger here is that emaging a bunch of vendors will flatten unsuspecting new players on accident, kinda like how using a bunch of dehydrated carps isnt gonna get anyone but the most vulnerable players who you werent aiming to kill... but eh, that's life.

Another potential use is to sneakily emag a vendor which is next to someone. The splat stun is LONG. If they are distracted enough it's enough to allow for a full kill. But... that's also probably fine.

**Comment by Sadboysuss** [comment]
> The big danger here is that emaging a bunch of vendors will flatten unsuspecting new players on accident.

- mfw antag item is dangerous

- mfw new players die to antags

- mfw new players die to rampant brand intelligence event

**Comment by DaCoolBoss** [comment]
I don't like it because it goes against what the emag's established purpose, which is breaking locks and overriding safeties. If someone swipes an emag on a vendor, they will expect access to the contraband menu, or free goodies, or removing ID requirements. This is more like uploading a virus to a vendor.

**Comment by SimplySmiley123** [comment]
> I don't like it because it goes against what the emag's established purpose, which is breaking locks and overriding safeties. If someone swipes an emag on a vendor, they will expect access to the contraband menu, or free goodies, or removing ID requirements. This is more like uploading a virus to a vendor.

Counterargument:
Bots and borgs exist and get the "turned into something pretty lethal" treatment.

Plus from in-game perspective, sentient vendors event exists and implies all vendors are capable of deciding to waddle around crushing people and that by default they don't do that. Emag removing safeties preventing that would fit your definition.

**Review by SmArtKar — CHANGES_REQUESTED** [review]
I think that this is a step in the right direction (E-MAG being a tool of sabotage, not a power up), but this indeed is quite strong. Maybe give it a change (40%?) to make vendors shoot items instead, like rampant AI does?

**Pull request closed without merging**